kasmvnc/run.sh

```diff
@@ -159,21 +154,29 @@ else
 fi
 
 # Coder port-forwarding from dashboard only supports HTTP
-sudo bash -c "cat > /etc/kasmvnc/kasmvnc.yaml <<EOF
+sudo tee /etc/kasmvnc/kasmvnc.yaml > /dev/null << EOF
 network:
   protocol: http
   websocket_port: ${PORT}
   ssl:
     require_ssl: false
   udp:
     public_ip: 127.0.0.1
-EOF"
+EOF
 
 # This password is not used since we start the server without auth.
 # The server is protected via the Coder session token / tunnel
 # and does not listen publicly
-echo -e "password\npassword\n" | vncpasswd -wo -u $USER
+echo -e "password\npassword\n" | vncpasswd -wo -u "$USER"
 
 # Start the server
 printf "🚀 Starting KasmVNC server...\n"
-sudo -u $USER bash -c "vncserver -select-de ${DESKTOP_ENVIRONMENT} -disableBasicAuth" > /tmp/kasmvncserver.log 2>&1 &
+# shellcheck disable=SC2024
+sudo -u "$USER" bash -c "vncserver -select-de ${DESKTOP_ENVIRONMENT} -disableBasicAuth" > /tmp/kasmvncserver.log 2>&1 &
```

Comment:
Handle for the case when there is no `sudo`. All `coder_script` resources are run by the user who runs the `coder_agent.*.init_script`, and it may not have passwordless sudo or no sudo at all. 

We can make these changes in a separate PR, too, but I recommend we handle them now, as this is a major refactor.

Reply:
Do we need sudo to start the `vncserver`?

Reply:
As far as I have tested no.